dace/codegen/cppunparse.py

```diff
@@ -313,9 +313,9 @@ def _Assign(self, t):
                 if self.type_inference is True:
                     # Perform type inference
                     # Build dictionary with symbols
-                    def_symbols = self.defined_symbols.copy()
-                    def_symbols.update(
-                        self.locals.get_name_type_associations())
+                    def_symbols = {}
+                    def_symbols.update(self.locals.get_name_type_associations())
+                    def_symbols.update(self.defined_symbols)
```

Comment:
what does that change? it seems like defined_symbols take precedence over self.locals?

Reply:
I had a case where locals contained memlets names set to `None`, which overwrote the real memlets passed by `defined_symbols`, so type inference failed. I don't know why they are in `locals`, or why they can be `None`.

Reply:
Strange. This will definitely resurface once we remove the extensive use of the `auto` keyword. I'm ok with this code then (for now).